packages/material-ui/src/Tabs/Tabs.test.js

```diff
@@ -53,9 +54,14 @@ describe('<Tabs />', () => {
     // expect(container.scrollLeft).to.equal(200); 💥
     if (isSafari) {
       this.skip();
+      global.ResizeObserver = ResizeObserver;
```

Comment:
We might want to silently deoptimize if the resize observer is not available. It would be a breaking change for developers that test the component in jsdom otherwise.

Reply:
Without importing a polyfill package on tests, I get a reference error since we do not have this API on Node. I don't think I fully understood. Can you elaborate on what I should do?

Reply:
>  I get a reference error since we do not have this API on Node

So will the developers that are testing the component in their own project. My proposal is to feature detect if ResizeObserver is available before using it.

Reply:
Instead of detecting in the test file itself, I imported polyfill and used try and catch in the component file itself for the reason mentioned here: https://github.com/mui-org/material-ui/pull/27791#issuecomment-899647694

The commit is here: https://github.com/mui-org/material-ui/commit/23651ff655d0e525c13406dab3574243f7d0041c#diff-8dd6f82db2612ff42348ed77f01b55934719b3dd3558de948af52aeedf030299R14

Reply:
Thanks. I created a mock resize observer so that there is no need to import or have a polyfill as a dependency. I also deleted the hook. https://github.com/mui-org/material-ui/pull/27791/commits/d591725a0a15d3e94b5969c305ac3d2c48c3439b